tool/internal/setup/extract.go

```diff
@@ -0,0 +1,143 @@
+// Copyright The OpenTelemetry Authors
+// SPDX-License-Identifier: Apache-2.0
+
+package setup
+
+import (
+	"archive/tar"
+	"compress/gzip"
+	"errors"
+	"io"
+	"os"
+	"path/filepath"
+	"strings"
+
+	"github.com/open-telemetry/opentelemetry-go-compile-instrumentation/tool/data"
+	"github.com/open-telemetry/opentelemetry-go-compile-instrumentation/tool/ex"
+	"github.com/open-telemetry/opentelemetry-go-compile-instrumentation/tool/util"
+)
+
+const (
+	EmbeddedInstPkgGzip = "otel-pkg.gz"
+)
+
+func normalizePath(name string) string {
+	const pkg, pkgTemp = "pkg", "pkg_temp"
+	cleanName := filepath.ToSlash(filepath.Clean(name))
+	if strings.HasPrefix(cleanName, pkgTemp+"/") {
+		cleanName = strings.Replace(cleanName, pkgTemp+"/", pkg+"/", 1)
+	} else if cleanName == pkgTemp {
+		cleanName = pkg
+	}
+	return cleanName
+}
+
+func extract(tarReader *tar.Reader, header *tar.Header, targetPath string) error {
+	fileInfo := header.FileInfo()
+	switch header.Typeflag {
+	case tar.TypeDir:
+		err := os.MkdirAll(targetPath, fileInfo.Mode())
+		if err != nil {
+			return ex.Error(err)
+		}
+
+	case tar.TypeReg:
+		{
+			file, err := os.OpenFile(targetPath, os.O_CREATE|os.O_RDWR,
+				fileInfo.Mode())
+			if err != nil {
+				return ex.Error(err)
+			}
+
+			_, err = io.CopyN(file, tarReader, header.Size)
+			if err != nil {
+				return ex.Error(err)
+			}
+			err = file.Close()
+			if err != nil {
+				return ex.Error(err)
+			}
+		}
+	default:
+		return ex.Errorf(nil, "unsupported file type: %c in %s",
+			header.Typeflag, header.Name)
+	}
+	return nil
+}
+
+func extractGZip(data []byte, targetDir string) error {
+	err0 := os.MkdirAll(targetDir, 0o755)
+	if err0 != nil {
+		return ex.Error(err0)
+	}
+
+	gzReader, err0 := gzip.NewReader(strings.NewReader(string(data)))
+	if err0 != nil {
+		return ex.Error(err0)
+	}
+	defer func() {
+		err0 = gzReader.Close()
+		if err0 != nil {
+			ex.Fatal(err0)
+		}
+	}()
+
+	tarReader := tar.NewReader(gzReader)
+	for {
+		header, err := tarReader.Next()
+		if errors.Is(err, io.EOF) {
+			break
+		}
+		if err != nil {
+			return ex.Error(err)
+		}
+
+		// Skip AppleDouble files (._filename) and other hidden files
+		if strings.HasPrefix(filepath.Base(header.Name), "._") ||
+			strings.HasPrefix(filepath.Base(header.Name), ".") {
```

Comment:
Strings that start with `._` all start with `.` so it is pointless to check both.

```suggestion
		if strings.HasPrefix(filepath.Base(header.Name), ".") {
```